metadata-utils/src/test/java/com/linkedin/metadata/utils/elasticsearch/IndexConventionImplTest.java

```diff
@@ -0,0 +1,37 @@
+package com.linkedin.metadata.utils.elasticsearch;
+
+import java.util.Optional;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+public class IndexConventionImplTest{
+
+  @Test
+  public void testIndexConventionNoPrefix() {
+    IndexConvention indexConventionNoPrefix = new IndexConventionImpl(null);
+    String entityName = "dataset";
+    String expectedIndexName = "datasetindex_v2";
+    assertEquals(indexConventionNoPrefix.getEntityIndexName(entityName), expectedIndexName);
+    assertEquals(indexConventionNoPrefix.getPrefix(), Optional.empty());
+    assertEquals(indexConventionNoPrefix.getEntityName(expectedIndexName), Optional.of(entityName));
+    assertEquals(indexConventionNoPrefix.getEntityName("totally not an index"), Optional.empty());
+    assertEquals(indexConventionNoPrefix.getEntityName("dataset_v2"), Optional.empty());
+    assertEquals(indexConventionNoPrefix.getEntityName("dashboardindex_v2_1683649932260"), Optional.of("dashboard"));
+  }
+
+  @Test
+  public void testIndexConventionPrefix() {
+    IndexConvention indexConventionPrefix = new IndexConventionImpl("prefix");
+    String entityName = "dataset";
+    String expectedIndexName = "prefix_datasetindex_v2";
+    assertEquals(indexConventionPrefix.getEntityIndexName(entityName), expectedIndexName);
+    assertEquals(indexConventionPrefix.getPrefix(), Optional.of("prefix"));
+    assertEquals(indexConventionPrefix.getEntityName(expectedIndexName), Optional.of(entityName));
```

Comment:
so right now we send back entity type values as the all caps, separated by underscores version. do you think we could keep that consistent? that's what the frontend usually expects from backend for an `EntityType` so we can convert it properly